### PR TITLE
Feat: Notifications

### PR DESCRIPTION
**Pull request by barshimi**
Close: #6657

### Summary
🚀🧹🧪

This pull request introduces a new package, `amplication-notification-service`, which implements the notification service using the Novu notification API and the Kafka message broker. It also refactors the existing packages, `amplication-build-manager` and `amplication-git-pull-request-service`, to use a common enum of Kafka topics from the `schema-registry` package, and updates some TypeScript and ESLint configurations for better code quality.

> _The project uses Kafka for messaging_
> _And NestJS for microservice-ing_
> _They added a `KAFKA_TOPICS` enum_
> _And a custom transport for them_
> _And Novu for notification dressing_

### Walkthrough
*  Add an enum of Kafka topics to the schema-registry package and use it in other packages to avoid hard-coding topic names ([link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-8fc1b1714e3919156dbc8fa42fbcf37fb24a23c3114c8d499e9185acb877fc13R11-R29), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-a040427a087e5dda6c8e9f8c2005037694d6e2c8b469818912a508e0458cb424L5-R5), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-a040427a087e5dda6c8e9f8c2005037694d6e2c8b469818912a508e0458cb424L39-R37), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-a040427a087e5dda6c8e9f8c2005037694d6e2c8b469818912a508e0458cb424L73-R71), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-243e5dc75c8a6ca029dac34dadf96afd121d56b8c6895888240190d8a1a3f4d2L6-R6), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-243e5dc75c8a6ca029dac34dadf96afd121d56b8c6895888240190d8a1a3f4d2L24-R24), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bR10), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bL59-R62), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bL108-R109), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bL142-R143), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bL168-R169), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bL194-R195), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bL285-R286), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17R16), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17L44-R44), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17L56-R56), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17L73-R73), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17L81-R81), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17L110-R108), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-7414ed5c548aedbceddde86507ee0c115f05b281894a43d5f5abe385001b1735L5-L17))
* Add a custom Kafka transport strategy for NestJS microservices and a unit test file for it ([link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-e6c583e815c92c135a8b593f2576109ecd446621700eb77ce2119ef9d21fd078R5), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-526ef8c7256b980db1a61f0b73bd6930bb529df7ddb50a7382e80105620ffa1dR1-R64), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-69649ae343cc1f72cae8402ad24bda6487c0e13d2beed1a178ebb42f4b5ffd28R1-R45))
* Install and integrate with the Novu notification service using the @novu/notification-center package and add a notification bell and a notification center UI to the workspace header component ([link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R62-R63), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-7fdb758d72b986ee99ae091fcd9f9b1312ecd063170e9dcbd9d99d5f47812d24R33-R38), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-7fdb758d72b986ee99ae091fcd9f9b1312ecd063170e9dcbd9d99d5f47812d24R127-R133), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-7fdb758d72b986ee99ae091fcd9f9b1312ecd063170e9dcbd9d99d5f47812d24R367-R382))
* Update the TypeScript compiler options to target es2022 and enable strict mode in the amplication-build-manager package ([link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-06cc88ca894ad3feaa026f9bf46c5aac8b72a229e821ec635157e8b32258682eL8-R9))
* Fix a typo in the test description of the build-runner.controller.spec.ts file ([link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0c8c0049207f0f2eb216ff7ef2a0a4279eb08309a6e955af38cb23024977790bL259-R260))
* Remove unused imports and environment variables from the amplication-build-manager and amplication-git-pull-request-service packages ([link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-ead4ab1598bcdb2b9593a43f5299b26787f4709777e29d06535eeca52d411333L9-L15), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-5c940f381b508c980f52d86bfc5418d7aa34d5c60f43628bfbd5a3459cf40d17L1), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-6ab196ec4fbb9df9a978658b4d59be5075bce91f89e22bcf369df3e6c66df8caL11-L18))
* Add a new package for the notification service with a .env file, a .eslintrc.json file, a .gitignore file, a Dockerfile, a jest.config.ts file, and a README.md file ([link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-0f7cb7328c0bad63ac4be46f5cd6d96435e9b1b3fed1b2e1b5001898a7677330L1-R9), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-b432f2f35088796de27181d78d3874ff2640b8ab7c6ef86442de5c2619696491R1-R56), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-f4efbdfbba7fcbe0164ec1f7e797c8bfc62fe16c5eddfc0490c3ff7135510426L1-R34), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-8c5bfcbf87762a092525db7a2295c746846c27e8ecd49596f6350ecd26d0070eL1-R13), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-1c80db3e65001c6f5f415b727174dd3bb6855d8d6093a81b11b92ebbdf81265eR1-R22), [link](https://github.com/amplication/amplication/pull/6698/files?diff=unified&w=0#diff-4d96b63f90af8ef256fae5b86db3fd6381f8b330b6bb7e79784501dde433ab22R1-R11))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-build-manager/src/build-logger/build-logger.controller.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class BuildLoggerController {
       value: logEntry,
     };
     await this.producerService.emitMessage(
-      this.configService.get(Env.DSG_LOG_TOPIC),
+      this.configService.get(KAFKA_TOPICS.DSG_LOG_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-logger/build-logger.controller.ts:24]: The topic name is now hardcoded in an enum and not loaded anymore from environment variables
```suggestion
      KAFKA_TOPICS,
```

File: packages/amplication-build-manager/src/build-logger/build-logger.controller.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { Body, Controller, Post } from "@nestjs/common";
 import { ConfigService } from "@nestjs/config";
 import { Env } from "../env";
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-logger/build-logger.controller.ts:4]: ConfigService and env are not used anymore.

Let's also remove the dependency from the constructor

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts [hunk 3]
```diff
@@ -105,7 +106,7 @@ describe("BuildRunnerController", () => {
     ).resolves.not.toThrow();
 
     expect(mockKafkaServiceEmitMessage).toBeCalledWith(
-      configService.get(Env.CODE_GENERATION_SUCCESS_TOPIC),
+      configService.get(KAFKA_TOPICS.CODE_GENERATION_SUCCESS_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts:109]: Same as here https://github.com/amplication/amplication/pull/6698/files#diff-243e5dc75c8a6ca029dac34dadf96afd121d56b8c6895888240190d8a1a3f4d2R24 
```suggestion
      KAFKA_TOPICS.CODE_GENERATION_SUCCESS_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts [hunk 4]
```diff
@@ -139,7 +140,7 @@ describe("BuildRunnerController", () => {
     ).rejects.toThrow(errorMock);
 
     expect(mockKafkaServiceEmitMessage).toBeCalledWith(
-      configService.get(Env.CODE_GENERATION_FAILURE_TOPIC),
+      configService.get(KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts:143]: https://github.com/amplication/amplication/pull/6698/files#diff-243e5dc75c8a6ca029dac34dadf96afd121d56b8c6895888240190d8a1a3f4d2R24

```suggestion
      CODE_GENERATION_FAILURE_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts [hunk 5]
```diff
@@ -165,7 +166,7 @@ describe("BuildRunnerController", () => {
     await controller.onCodeGenerationFailure(codeGenerationFailureDTOMock);
 
     expect(mockKafkaServiceEmitMessage).toBeCalledWith(
-      configService.get(Env.CODE_GENERATION_FAILURE_TOPIC),
+      configService.get(KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts:169]: Same as https://github.com/amplication/amplication/pull/6698/files#diff-243e5dc75c8a6ca029dac34dadf96afd121d56b8c6895888240190d8a1a3f4d2R24
```suggestion
      KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts [hunk 6]
```diff
@@ -282,7 +283,7 @@ describe("BuildRunnerController", () => {
 
     expect(loggerService.error).toBeCalledWith(errorMock.message, errorMock);
     expect(mockKafkaServiceEmitMessage).toBeCalledWith(
-      configService.get(Env.CODE_GENERATION_FAILURE_TOPIC),
+      configService.get(KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts:286]: Same as https://github.com/amplication/amplication/pull/6698/files#diff-243e5dc75c8a6ca029dac34dadf96afd121d56b8c6895888240190d8a1a3f4d2R24
```suggestion
     KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts [hunk 7]
```diff
@@ -191,7 +192,7 @@ describe("BuildRunnerController", () => {
     await controller.onCodeGenerationFailure(codeGenerationFailureDTOMock);
 
     expect(mockKafkaServiceEmitMessage).toBeCalledWith(
-      configService.get(Env.CODE_GENERATION_FAILURE_TOPIC),
+      configService.get(KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts:195]: Same as https://github.com/amplication/amplication/pull/6698/files#diff-243e5dc75c8a6ca029dac34dadf96afd121d56b8c6895888240190d8a1a3f4d2R24
```suggestion
      KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.ts [hunk 8]
```diff
@@ -41,7 +41,7 @@ export class BuildRunnerController {
       };
 
       await this.producerService.emitMessage(
-        this.configService.get(Env.CODE_GENERATION_SUCCESS_TOPIC),
+        this.configService.get(KAFKA_TOPICS.CODE_GENERATION_SUCCESS_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.ts:44]:
```suggestion
       KAFKA_TOPICS.CODE_GENERATION_SUCCESS_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.ts [hunk 9]
```diff
@@ -53,7 +53,7 @@ export class BuildRunnerController {
       };
 
       await this.producerService.emitMessage(
-        this.configService.get(Env.CODE_GENERATION_FAILURE_TOPIC),
+        this.configService.get(KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC),
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.ts:56]:
```suggestion
       KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.ts [hunk 10]
```diff
@@ -70,17 +70,15 @@ export class BuildRunnerController {
...
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.ts:73]:
```suggestion
        KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC,
```

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.ts [hunk 11]
```diff
@@ -107,7 +105,7 @@ export class BuildRunnerController {
...
```
Comment (overbit) [inline packages/amplication-build-manager/src/build-runner/build-runner.controller.ts:108]:
```suggestion
        KAFKA_TOPICS.CODE_GENERATION_FAILURE_TOPIC,
```

File: packages/amplication-notification-service/.env [hunk 12]
```diff
@@ -0,0 +1,12 @@
+PORT=5010
+
+# Kafka
+KAFKA_BROKERS="localhost:9092"
+KAFKA_CLIENT_ID='build-manager'
+KAFKA_GROUP_ID='build-manager'
```
Comment (overbit) [inline packages/amplication-notification-service/.env:6]: be careful with the naming

File: packages/amplication-notification-service/.env [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+KAFKA_CLIENT_CONFIG_SSL='false'
+
+DSG_RUNNER_URL="http://localhost:8900/api/code-generate"
+
+# Default env
+DSG_JOBS_RESOURCE_DATA_FILE="input.json"
```
Comment (overbit) [inline packages/amplication-notification-service/.env:12]: Potential copy mistake? If not relevant let's remove these

File: packages/amplication-notification-service/src/app.service.ts [hunk 14]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from "@nestjs/common";
+
+@Injectable()
+export class AppService {
+  getHello(): string {
+    return "Hello World!";
+  }
```
Comment (overbit) [inline packages/amplication-notification-service/src/app.service.ts:7]: if not needed, let's remove this service (and the controller)

File: packages/amplication-notification-service/src/env.ts [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+  static readonly PORT = "PORT";
+
+  static readonly DSG_RUNNER_URL = "DSG_RUNNER_URL";
+
+  static readonly BUILD_ARTIFACTS_BASE_FOLDER = "BUILD_ARTIFACTS_BASE_FOLDER";
+  static readonly BUILD_ARTIFACTS_CODE_FOLDER = "BUILD_ARTIFACTS_CODE_FOLDER";
+
+  static readonly DSG_JOBS_BASE_FOLDER = "DSG_JOBS_BASE_FOLDER";
+  static readonly DSG_JOBS_CODE_FOLDER = "DSG_JOBS_CODE_FOLDER";
+  static readonly DSG_JOBS_RESOURCE_DATA_FILE = "DSG_JOBS_RESOURCE_DATA_FILE";
```
Comment (overbit) [inline packages/amplication-notification-service/src/env.ts:12]: if not needed, let's remove these

File: packages/amplication-notification-service/src/env.ts [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+
+  static readonly SERVICE_NAME = "amplication-build-manager";
```
Comment (overbit) [inline packages/amplication-notification-service/src/env.ts:14]: watch out for the service name 👀

File: packages/amplication-notification-service/src/test.spec.ts [hunk 17]
```diff
@@ -0,0 +1,5 @@
+describe("", () => {
+  it("", () => {
+    expect(true);
+  });
+});
```
Comment (overbit) [inline packages/amplication-notification-service/src/test.spec.ts:5]: if not needed, let's remove this redundant test file

File: packages/amplication-notification-service/tsconfig.app.json [hunk 18]
```diff
@@ -0,0 +1,12 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../dist/out-tsc",
+    "module": "commonjs",
+    "types": ["node"],
+    "emitDecoratorMetadata": true,
+    "target": "es2015"
```
Comment (overbit) [inline packages/amplication-notification-service/tsconfig.app.json:8]: Since it is a new project, let's try to start with the right foot. 🦶 

```suggestion
    "target": "es2022",
    "strict": true,
```

File: packages/amplication-server/src/core/build/build.service.ts [hunk 19]
```diff
@@ -371,7 +372,7 @@ export class BuildService {
         };
 
         await this.kafkaProducerService.emitMessage(
-          this.configService.get(Env.CODE_GENERATION_REQUEST_TOPIC),
+          this.configService.get(KAFKA_TOPICS.CODE_GENERATION_REQUEST_TOPIC),
```
Comment (overbit) [inline packages/amplication-server/src/core/build/build.service.ts:375]:
```suggestion
          KAFKA_TOPICS.CODE_GENERATION_REQUEST_TOPIC,
```

File: packages/amplication-server/src/core/build/build.service.ts [hunk 20]
```diff
@@ -693,7 +694,7 @@ export class BuildService {
             value: createPullRequestMessage,
           };
           await this.kafkaProducerService.emitMessage(
-            this.configService.get(Env.CREATE_PR_REQUEST_TOPIC),
+            this.configService.get(KAFKA_TOPICS.CREATE_PR_REQUEST_TOPIC),
```
Comment (overbit) [inline packages/amplication-server/src/core/build/build.service.ts:697]:
```suggestion
           KAFKA_TOPICS.CREATE_PR_REQUEST_TOPIC,
```

File: packages/amplication-server/src/core/dbSchemaImport/dbSchemaImport.service.ts [hunk 21]
```diff
@@ -54,7 +56,7 @@ export class DBSchemaImportService {
     };
 
     await this.kafkaProducerService.emitMessage(
-      this.configService.get(Env.DB_SCHEMA_IMPORT_TOPIC),
+      this.configService.get(KAFKA_TOPICS.DB_SCHEMA_IMPORT_TOPIC),
```
Comment (overbit) [inline packages/amplication-server/src/core/dbSchemaImport/dbSchemaImport.service.ts:59]:
```suggestion
      KAFKA_TOPICS.DB_SCHEMA_IMPORT_TOPIC,
```

File: packages/amplication-server/src/core/dbSchemaImport/dbSchemaImport.service.ts [hunk 22]
```diff
@@ -99,7 +101,7 @@ export class DBSchemaImportService {
         const actionContext = this.actionService.createActionContext(
           dbSchemaImportAction.id,
           step,
-          this.configService.get(Env.USER_ACTION_LOG_TOPIC)
+          this.configService.get(KAFKA_TOPICS.USER_ACTION_LOG_TOPIC)
```
Comment (overbit) [inline packages/amplication-server/src/core/dbSchemaImport/dbSchemaImport.service.ts:104]:
```suggestion
          KAFKA_TOPICS.USER_ACTION_LOG_TOPIC
```

**Review by overbit — CHANGES_REQUESTED** [review]

File: libs/util/nestjs/kafka/src/kafka.transport.ts
Comment (overbit) [inline libs/util/nestjs/kafka/src/kafka.transport.ts:1]: Let's add some basic unit tests for this transport:

```ts
import { MessageHandler } from "@nestjs/microservices";
import { KafkaCustomTransport } from "./kafka.transport"; // Replace with correct path
import { Consumer } from "@nestjs/microservices/external/kafka.interface";

describe("KafkaCustomTransport", () => {
  let kafkaCustomTransport: KafkaCustomTransport;
  let mockSubscribe: jest.SpyInstance;
  let mockRun: jest.SpyInstance;

  beforeEach(() => {
    jest.clearAllMocks();
    kafkaCustomTransport = new KafkaCustomTransport({});
  });

  it("should bind events correctly", async () => {
    mockSubscribe = jest.fn();
    mockRun = jest.fn();

    const mockConsumer: Consumer = {
      subscribe: mockSubscribe,
      run: mockRun,
    } as unknown as Consumer;

    const pattern = "test-pattern";
    const regexp = "/test/";
    kafkaCustomTransport["messageHandlers"].set(pattern, <MessageHandler>{});
    kafkaCustomTransport["messageHandlers"].set(regexp, <MessageHandler>{});

    await kafkaCustomTransport.bindEvents(mockConsumer);

    expect(mockSubscribe).toHaveBeenCalled();
    expect(mockRun).toHaveBeenCalled();
  });

  it("should get handler by pattern or RegExp", () => {
    const patternHandler = jest.fn();
    const regexpHandler = jest.fn();

    const pattern = "test-pattern";
    const regexp = "/test/";

    kafkaCustomTransport["messageHandlers"].set(
      pattern,
      <MessageHandler>patternHandler
    );
    kafkaCustomTransport["messageHandlers"].set(
      regexp,
      <MessageHandler>regexpHandler
    );

    const patternResult = kafkaCustomTransport.getHandlerByPattern(pattern);
    const regexpResult = kafkaCustomTransport.getHandlerByPattern(regexp);

    expect(patternResult).toBe(patternHandler);
    expect(regexpResult).toBe(regexpHandler);
  });

  it("should return null when no handler is found", () => {
    const result = kafkaCustomTransport.getHandlerByPattern(
      "non-existing-pattern"
    );
    expect(result).toBeNull();
  });
});

```

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-server/src/env.ts [hunk 23]
```diff
@@ -17,23 +17,6 @@ export class Env {
...
-  static readonly CREATE_PR_REQUEST_TOPIC = "CREATE_PR_REQUEST_TOPIC";
-  static readonly CREATE_PR_SUCCESS_TOPIC = "CREATE_PR_SUCCESS_TOPIC";
-  static readonly CREATE_PR_FAILURE_TOPIC = "CREATE_PR_FAILURE_TOPIC";
-
-  static readonly DSG_LOG_TOPIC = "DSG_LOG_TOPIC";
-
-  static readonly DB_SCHEMA_IMPORT_TOPIC = "DB_SCHEMA_IMPORT_TOPIC";
-  static readonly USER_ACTION_LOG_TOPIC = "USER_ACTION_LOG_TOPIC";
-  static readonly USER_ACTION_TOPIC = "USER_ACTION_TOPIC";
-
```
Comment (abrl91) [inline packages/amplication-server/src/env.ts:36]: Why did you remove these environment variables?

File: packages/amplication-server/src/util/encryptionUtil.ts [hunk 24]
```diff
@@ -0,0 +1,3 @@
+import MD5 from "crypto-js/md5";
+
+export const encryptString = (str: string) => MD5(str).toString();
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-server/src/util/encryptionUtil.ts:3]: @barshimi  have a look at this

File: libs/schema-registry/src/lib/user-action/key.ts [hunk 25]
```diff
@@ -1,6 +1 @@
-import { IsString } from "class-validator";
-
-export class Key {
-  @IsString()
-  userId!: string;
-}
+export class Key {}
```
Comment (abrl91) [inline libs/schema-registry/src/lib/user-action/key.ts:1]: we don't need the key?

File: packages/amplication-server/src/core/auth/auth.service.ts [hunk 26]
```diff
@@ -59,7 +56,7 @@ export class AuthService {
     private readonly passwordService: PasswordService,
     private readonly prismaService: PrismaService,
     private readonly accountService: AccountService,
-    private readonly kafkaProducerService: KafkaProducerService,
+    // private readonly kafkaProducerService: KafkaProducerService,
```
Comment (abrl91) [inline packages/amplication-server/src/core/auth/auth.service.ts:59]: please remove it if we don't need it

File: packages/amplication-server/src/util/encryptionUtil.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (barshimi) [reply packages/amplication-server/src/util/encryptionUtil.ts:3]: please resolve it !!

File: packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts (hunk 7 above)
Comment (abrl91) [reply packages/amplication-build-manager/src/build-runner/build-runner.controller.spec.ts:195]: @barshimi ?

File: packages/amplication-notification-service/src/app.service.ts [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+import { AmplicationLogger } from "@amplication/util/nestjs/logging";
+import { NovuService } from "./util/novuService";
+import { novuPackage } from "./util/novuPackage";
+import { subscribeUser } from "./notification-packages/subscribeUser";
+import { buildCompleted } from "./notification-packages/buildCompleted";
+
+const compose =
+  (...fns) =>
+  (x) =>
+    fns.reduce(async (y, f) => f(await y), x);
```
Comment (overbit) [inline packages/amplication-notification-service/src/app.service.ts:11]: Let's try to add some unit test here to get confidence in the solution

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-notification-service/src/app.service.ts (hunk 27 above)
Comment (barshimi) [reply packages/amplication-notification-service/src/app.service.ts:11]: It's a simple composition, nothing to test here

File: packages/amplication-server/src/util/encryptionUtil.ts (hunk 24 above)
[the remark this answers is not in the document]
Comment (overbit) [reply packages/amplication-server/src/util/encryptionUtil.ts:3]: Can we use a stronger cryptographic algorithm  here?
